### PR TITLE
add hardhat-interface-generator package

### PR DESCRIPTION
**Pull request by julianmrodri**
* Solves: #245 
* Adds the `hardhat-interface-generator` package that allows you to create interfaces based from a contract.
* Adds the required command so you can run `yarn gen-interface`
```bash
$ yarn gen-interface "YourContract"
```

```
yarn run v1.22.5
$ yarn workspace @scaffold-eth/hardhat gen-interface YourContract
$ hardhat gen-interface YourContract
Generated interface IYourContract
✨  Done in 1.61s.
```

### CODE REVIEW

File: packages/hardhat/package.json [hunk 1]
```diff
@@ -38,6 +39,7 @@
     "balance": "hardhat balance",
     "send": "hardhat send",
     "generate": "hardhat generate",
+    "gen-interface": "hardhat gen-interface",
```
Comment (codenamejason) [inline packages/hardhat/package.json:42]: rename to "interface".

**Review by codenamejason — CHANGES_REQUESTED** [review]
Everything looks good, I tested it on my local with no issues. Just need to rename it to "interface" and we are all good.

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "s3": "yarn workspace @scaffold-eth/react-app s3",
     "ship": "yarn workspace @scaffold-eth/react-app ship",
     "generate": "yarn workspace @scaffold-eth/hardhat generate",
+    "gen-interface": "yarn workspace @scaffold-eth/hardhat gen-interface",
```
[the remark this answers is not in the document]
Comment (julianmrodri) [reply package.json]: Done.

File: packages/hardhat/package.json (hunk 1 above)
Comment (julianmrodri) [reply packages/hardhat/package.json]: Done

**Review by codenamejason — APPROVED** [review]

**Comment by StErMi** [comment]
Yesterday @austintgriffith suggested to me this package and I was not totally sold by it. It's maybe fine to generate the interface of a "simple" contract but in my case where the contract was extending OpenZeppelin contracts, it generated a "faulty" interface that it was not usable. 

I had to delete it and write it manually by myself. I'll copy what I've said to @austintgriffith as a feedback

> I’ve tried it but it’s not good or at least not good for a complex contract like this that relies on external. it does not rely on OpenZeppelin interface and the code generated does not compile very well (a lot of warnings) because of the overloaded method, he does not add the SPDX-License and he does not add the pragma

**Pull request closed without merging**